code/modules/admin/poll_management.dm

```diff
@@ -0,0 +1,735 @@
+		poll.admin_only = TRUE
+	else
+		poll.admin_only = FALSE
+	if(href_list["dontshow"])
+		poll.dont_show = TRUE
+	else
+		poll.dont_show = FALSE
+	if(href_list["allowrevoting"])
+		poll.allow_revoting = TRUE
+	else
+		poll.allow_revoting = FALSE
+	if(href_list["clearvotesedit"])
+		clear_votes = TRUE
+	if(href_list["submitpoll"])
+		submit_ready = TRUE
+	if(poll.poll_type == POLLTYPE_MULTI)
+		if(text2num(href_list["optionsallowed"]))
+			poll.options_allowed = text2num(href_list["optionsallowed"])
+			if(poll.options_allowed == 1)
+				error_state += "Multiple choice polls require more than one option allowed, use a standard option poll for singlular voting."
+			if(poll.options_allowed < 0)
+				error_state += "Multiple choice options allowed cannot be negative."
+		else
+			error_state += "Multiple choice poll was selected but no number of allowed options was provided."
+	if(submit_ready && poll.poll_type != POLLTYPE_TEXT && !length(poll.options))
+		error_state += "This poll type requires at least one option."
+	if(error_state.len)
+		if(poll.edit_ready)
+			to_chat(usr, "<span class='danger'>Not all edits were applied because the following errors were present:\n[error_state.Join("\n")]</span>")
+		else
+			to_chat(usr, "<span class='danger'>Poll not [new_poll ? "initialized" : "submitted"] because the following errors were present:\n[error_state.Join("\n")]</span>")
+			if(new_poll)
+				qdel(poll)
+		return
+	if(submit_ready)
+		var/db = poll.edit_ready //if the poll is new it will need its options inserted for the first time
+		poll.save_poll_data(clear_votes)
+		if(!db)
+			poll.save_all_options()
+	poll_management_panel(poll)
+
+/datum/poll_question/New(id, polltype, starttime, endtime, question, subtitle, adminonly, multiplechoiceoptions, dontshow, allow_revoting, vote_count, creator, future, dbload = FALSE)
+	poll_id = text2num(id)
+	poll_type = polltype
+	start_datetime = starttime
+	end_datetime = endtime
+	src.question = question
+	src.subtitle = subtitle
+	admin_only = text2num(adminonly)
+	options_allowed = text2num(multiplechoiceoptions)
+	dont_show = text2num(dontshow)
+	src.allow_revoting = text2num(allow_revoting)
+	poll_votes = text2num(vote_count) || 0
+	created_by = creator
+	future_poll = text2num(future)
+	edit_ready = dbload
+	GLOB.polls += src
+
+/datum/poll_question/Destroy()
+	GLOB.polls -= src
+	return ..()
+
+/**
+  * Sets a poll and its associated data as deleted in the database.
+  *
+  * Calls the procedure set_poll_deleted to set the deleted column to 1 for each row in the poll_ tables matching the poll id used.
+  * Then deletes each option datum and finally the poll itself.
+  *
+  */
+/datum/poll_question/proc/delete_poll()
+	if(!check_rights(R_POLL))
+		return
+	if(!SSdbcore.Connect())
+		to_chat(usr, "<span class='danger'>Failed to establish database connection.</span>")
+		return
+	var/datum/DBQuery/query_delete_poll = SSdbcore.NewQuery("CALL set_poll_deleted('[sanitizeSQL(poll_id)]')")
```

Comment:
You could probably do with marking poll data for reloading if server A deletes a poll then someone on server B tries to vote on it since there's no explicit synchro.

Reply:
Thought about it but didn't actually try, I'll have a look at how this can be done with world topic.